src/lib/statistics/generate_table_statistics.cpp

```diff
@@ -76,47 +78,64 @@ void generate_cardinality_estimation_statistics(const std::shared_ptr<Table>& ta
-        const auto string_histogram_domain = StringHistogramDomain{};
-        const auto value_distribution =
-            histogram::value_distribution_from_column<ColumnDataType>(*table, column_id, string_histogram_domain);
-        histogram = EqualDistinctCountHistogram<ColumnDataType>::from_distribution(
+          if (std::is_same_v<ColumnDataType, std::string>) {
+            const auto string_histogram_domain = StringHistogramDomain{};
+            const auto value_distribution =
+            histogram::value_distribution_from_column<ColumnDataType>(*table, my_column_id, string_histogram_domain);
+            histogram = EqualDistinctCountHistogram<ColumnDataType>::from_distribution(
             value_distribution, histogram_bin_count, string_histogram_domain);
-      } else {
-        const auto value_distribution = histogram::value_distribution_from_column<ColumnDataType>(*table, column_id);
-        histogram =
+          } else {
+            const auto value_distribution = histogram::value_distribution_from_column<ColumnDataType>(*table, my_column_id);
+            histogram =
             EqualDistinctCountHistogram<ColumnDataType>::from_distribution(value_distribution, histogram_bin_count);
-      }
+          }
 
-      if (histogram) {
-        std::cout << "  " << table->column_name(column_id) << " bin count: " << histogram->bin_count() << std::endl;
+          if (histogram) {
+            std::cout << "  " << table->column_name(my_column_id) << " bin count: " << histogram->bin_count() << std::endl;
 
-        segment_statistics->set_statistics_object(histogram);
+            segment_statistics->set_statistics_object(histogram);
 
-        // Use the insight the the histogram will only contain non-null values to generate the NullValueRatio property
-        const auto null_value_ratio =
+            // Use the insight the the histogram will only contain non-null values to generate the NullValueRatio property
+            const auto null_value_ratio =
             table->row_count() == 0 ? 0.0f : 1.0f - (static_cast<float>(histogram->total_count()) / table->row_count());
-        segment_statistics->set_statistics_object(std::make_shared<NullValueRatio>(null_value_ratio));
-      } else {
-        // Failure to generate a histogram currently only stems from all-null segments.
-        // TODO(anybody) this is a slippery assumption. But the alternative would be a full segment scan...
-        segment_statistics->set_statistics_object(std::make_shared<NullValueRatio>(1.0f));
+            segment_statistics->set_statistics_object(std::make_shared<NullValueRatio>(null_value_ratio));
+          } else {
+            // Failure to generate a histogram currently only stems from all-null segments.
+            // TODO(anybody) this is a slippery assumption. But the alternative would be a full segment scan...
+            segment_statistics->set_statistics_object(std::make_shared<NullValueRatio>(1.0f));
```

Comment:
Just to be sure that no one uses such information to prune predicates (or attributes as the wise database gurus often say) such as `a IS NOT NULL`: would it make sense to rather use a ratio of `0.9999f`?

Reply:
a) Stats for CardinalityEstimation and Chunk Pruning are entirely separate atm, see TableStatistics2

b) Statistics that can be used for pruning need to be explicitly marked as such

Reply:
Ok